### PR TITLE
[223] Add Degree confirmation component

### PR DESCRIPTION
**Pull request by aliuk2012**
### Changes proposed in this pull request
- Needed to pull in some work from ticket 224, please ignore first commit
- Adds new degrees confirmation component.
- we still need to add proper paths to the actions but that will be added in another PR once we have created those pages

### Guidance to review

- Visit https://dfe-twd-rttd-pr-100.herokuapp.com/view_components/trainees/confirmation/degrees/view
- Try out each example https://dfe-twd-rttd-pr-100.herokuapp.com/view_components/trainees/confirmation/degrees/view

### CODE REVIEW

**Comment by aliuk2012** [comment]
[[M] Build "Confirm degree" details](https://trello.com/c/2Grb8NIT/223-m-build-confirm-degree-details)

**Review by edwardhorsford — APPROVED** [review]
Looking good!

File: app/components/trainees/confirmation/degrees/view.html.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<% if trainee.degrees.present? %>
+  <% trainee.degrees.each do |degree| %>
+    <%= render_component SummaryCard::View.new(title: degree_title(degree),
```
Comment (darokel) [inline app/components/trainees/confirmation/degrees/view.html.erb:3]: can we break a new line after the first `(` so `title:` is on the next line

File: app/components/trainees/confirmation/degrees/view.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
+module Trainees
+  module Confirmation
+    module Degrees
+      class View < GovukComponent::Base
+        attr_accessor :trainee
+
+        def initialize(trainee:)
+          @trainee = trainee
```
Comment (darokel) [inline app/components/trainees/confirmation/degrees/view.rb:8]: Does this component even need the trainee? Looks like it's just interrogating a degree object.

Comment (aliuk2012) [reply app/components/trainees/confirmation/degrees/view.rb:8]: 🤔 Good point. I suppose at the time I was thinking just pass it the main thing i.e the trainee and let it work out which parts of it to use.

What do you reckon, drop trainee and just pass it a list of degrees?

Comment (aliuk2012) [reply app/components/trainees/confirmation/degrees/view.rb]: `trainee` is now dropped in favour of passing in a collection of degrees to the component.

**Review by darokel — APPROVED** [review]

**Pull request merged**